### PR TITLE
Add Weights and Biases to `MetricLogger` Class

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
This PR aims to add basic [Weights and Biases](https://wandb.ai/site) Metric Logging by appending to the existing MetricLogger Class defined in [logger.py](https://github.com/zhiqwang/yolov5-rt-stack/blob/main/yolort/utils/logger.py) with minimal changes while supporting Multiple GPU logging with torch distributed.

The changes can be summarized as follows :-

* Pass the `args` to the `MetricLogger` which if the `--use_wandb` is set to `True`, creates and run and logs metrics using the `update` function.
* Add 3 extra arguments namely `--use_wandb`, `--wandb_project` and `--wandb_entity` which can be used to specify whether to use wandb, the name of the project to be used ("yolov5-rt" by default) and name of the entity to be used.

### CODE REVIEW

File: tools/eval_metric.py [hunk 1]
```diff
@@ -146,9 +152,9 @@ def eval_metric(args):
 
 
 @torch.no_grad()
-def evaluate(model, data_loader, coco_evaluator, device, print_freq):
+def evaluate(model, data_loader, coco_evaluator, device, print_freq, args):
     # COCO evaluation
-    metric_logger = MetricLogger(delimiter="  ")
+    metric_logger = MetricLogger(args, delimiter="  ")
```
Comment (zhiqwang) [inline tools/eval_metric.py:157]: Hi @SauravMaheshkar , Could we pass in the specific parameters to be used instead of the args here?

File: yolort/utils/logger.py [hunk 2]
```diff
@@ -70,16 +77,21 @@ def __str__(self):
 
 
 class MetricLogger:
-    def __init__(self, delimiter="\t"):
+    def __init__(self, args, delimiter="\t"):
```
Comment (zhiqwang) [inline yolort/utils/logger.py:80]: Same as above

**Review by zhiqwang — COMMENTED** [review]
Hi @SauravMaheshkar , The overall looks great to me, I only have one comment on the parameter passing mechanism here, Let me know your thoughts here?

**Comment by SauravMaheshkar** [comment]
I prefer to keep `args` as it is passed as the value for the `config` argument in the `wandb.init()` statement viz.

```python
self.wandb_run = wandb.init(project=args.wandb_project, entity=args.wandb_entity, config=args)
```

using an extensive config is extremely beneficial for experiment tracking as it allows someone to filter, group and sort by a particular hyperparameter. For more details you can refer to the [**official documentation**](https://docs.wandb.ai/guides/track/config).

**Comment by zhiqwang** [comment]
> using an extensive config is extremely beneficial for experiment tracking as it allows someone to filter, group and sort by a particular hyperparameter. For more details you can refer to the [**official documentation**](https://docs.wandb.ai/guides/track/config).

Hi @SauravMaheshkar , Got it and thanks for providing the contextual information here.

In the current development of yolort project, we provide two main ways to call it, one is through the command line tool and the other is through direct package calls. In the first way, `argparse` is really powerful and flexible. For the second package approach, argparse can cause some difficulties when calling it.  In particular, the parsing of argparse becomes difficult when we use jupyter notebook,  you can check out this [problem](https://stackoverflow.com/questions/30656777/how-to-call-module-written-with-argparse-in-ipython-notebook) from a few years ago, and by now there is not really a particularly flexible solution to it.

`jupyter notebook` is a [literate programming](https://en.wikipedia.org/wiki/Literate_programming) paradigm, and there are some limitations to use it, I currently feel that it might be more appropriate if we take a different approach than argparse here.

**Comment by SauravMaheshkar** [comment]
Makes sense. For now I'll drop the `config` argument from the `wandb.init()` statement, and add the `wandb_project` and `wandb_entity` as a parameter to the function call.

File: tools/eval_metric.py [hunk 3]
```diff
@@ -146,9 +161,9 @@ def eval_metric(args):
 
 
 @torch.no_grad()
-def evaluate(model, data_loader, coco_evaluator, device, print_freq):
+def evaluate(model, data_loader, coco_evaluator, device, print_freq, use_wandb, wandb_project, wandb_entity):
     # COCO evaluation
-    metric_logger = MetricLogger(delimiter="  ")
+    metric_logger = MetricLogger(args, delimiter="  ")
```
Comment (zhiqwang) [inline tools/eval_metric.py:166]: Hi @SauravMaheshkar , Seems this `args` are forgotten to be adapted?

Comment (SauravMaheshkar) [reply tools/eval_metric.py]: Sorry missed that. Fixed in [a7a46f0](https://github.com/zhiqwang/yolov5-rt-stack/pull/426/commits/a7a46f06184684b2bcddbfc6c5cc9396007e74fa)

File: tools/eval_metric.py [hunk 4]
```diff
@@ -146,9 +161,9 @@ def eval_metric(args):
 
 
 @torch.no_grad()
-def evaluate(model, data_loader, coco_evaluator, device, print_freq):
+def evaluate(model, data_loader, coco_evaluator, device, print_freq, use_wandb, wandb_project, wandb_entity):
     # COCO evaluation
-    metric_logger = MetricLogger(delimiter="  ")
+    metric_logger = MetricLogger(use_wandb, wandb_project, wandb_entity, delimiter="  ")
```
Comment (zhiqwang) [inline tools/eval_metric.py:166]:
```suggestion
    metric_logger = MetricLogger(delimiter="  ", use_wandb=use_wandb, wandb_project=wandb_project, wandb_entity=wandb_entity)
```
Maybe we should adjust the order here.

**Review by zhiqwang — APPROVED** [review]
Thanks @SauravMaheshkar , We will integrate `Weights and Biases` for more functions, and let's wait GH actions to test the other parts.
